### ISSUE TITLE
protected method `method_name' called for ActionView::Template

### ISSUE THREAD

**Issue by TheKidCoder** [rails/rails#24309]
### Steps to reproduce
Render template using Rails 4.2.6

### Expected behavior
Template should have rendered.

### Actual behavior
Exception was thrown.
```
ActionView::Template::Error (protected method `method_name' called for app/views/users/index.haml:ActionView::Template):
actionview (4.2.6) lib/action_view/template.rb:269:in `compile' 
...
```

### System configuration
**Rails version**: 4.2.6

**Ruby version**: jRuby 9.0.5.0

The problem is coming from using `self.method_name` instead of just `method_name` because it is a protected method. This may be a jRuby problem, but I can't see a good reason to use `self`.

I'm putting together a patch to fix this.

**Comment by batter** [comment]
@TheKidCoder: can you provide more detailed instructions as to how to reproduce this? I can't seem to reproduce it, granted I am using normal MRI Ruby, so perhaps it is indeed jRuby specific.

**Comment by TheKidCoder** [comment]
This was fixed for me by upgrading to jruby-9.1.2.0.
